### ISSUE TITLE
Colon rule is wrongly applied to Dictionary constructor

### ISSUE THREAD

**Issue by inamiy** [realm/SwiftLint#891]
```swift
func hello<Msg>()
{
    let int1a = Int(1)                  // Int as identifier
    let int1b: Int = 1                  // Int as typeidentifier
    let dict1a = [String : Int]()       // String as identifier, Int as identifier
    let dict1b: [String : Int] = [:]    // String as typeidentifier, Int as typeidentifier

    // colon rule violation!
    let dict2a = [String : Msg]()       // String as identifier, Msg as typeidentifier

    let dict2b: [String : Msg] = [:]    // String as typeidentifier, Int as typeidentifier
}
```

In current version 0.13.0, above `dict2a` gets "colon rule violation warning" because its rule is to check "identifier" followed by "typeidentifier".
I'm not sure the right SyntaxToken of `[String : Msg]()`, but is this SourceKit(ten) bug?
And if this needs a workaround to solve, is it wise to also check first capital letter (treating it as type) to skip colon rule?

BTW, I can workaround this by writing `dict2b` style instead, so this issue is very trivial. 🙂

**Comment by marcelofabri** [comment]
Closing this for now since all these declarations should trigger 😬
